wordpress-mobile/WordPress-iOS
# Adds blogging reminders to site settings.

Adds blogging goals to settings.

Migrates some legacy code to Swift.

## To test:

You should make sure to test all of the options under Site Settings > General.
You should make sure to test these using both an iPad and an iPhone.
When checking the values are updated in the web, I recommend going into WP Admin > Settings > General, which is where all of these settings are stored.

### Site Title:

- Make sure you can edit the site title
- Make sure the edits can be seen on the web
- Make sure when exiting the editing screen, the cell value is properly updated

### Tagline:

- Make sure you can edit the tagline
- Make sure the edits can be seen on the web
- Make sure when exiting the editing screen, the cell value is properly updated

### Address:

- This should not be editable or selectable in any way.

### Privacy

- Make sure you can edit the privacy setting
- Make sure changing privacy settings is reflected on the web
- Make sure when exiting the privacy setting picker screen, the cell value is properly updated

### Language

- Make sure you can edit the language setting
- Make sure changing language settings is reflected on the web
- Make sure when exiting the language setting picker screen, the cell value is properly updated

### Timezone

- Make sure you can edit the timezone setting
- Make sure changing timezone settings is reflected on the web
- Make sure when exiting the timezone setting picker screen, the cell value is properly updated

### Blogging Reminders

- You should just see a cell with value "Undefined" on it.
- Make sure the cell isn't visible when you switch the feature flag off.
- Tapping on this cell should launch the blogging reminders flow.

## Regression Notes

1. Potential unintended areas of impact

To be able to add a new option under site settings, I needed to make some bigger changes to how we were managing the data for the table view.  I decided the easiest and cleanest solution was to switch some of the legacy code to Swift.

This means there's a change some of the old code could be broken.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

I tested all of the changed settings manually, and wrote a step by step to verify them during the review of this PR.

3. What automated tests I added (or what prevented me from doing so)

None

## PR submission checklist:
